packages/ui/src/helpers/authenticator/formFields/util.ts

```diff
@@ -0,0 +1,41 @@
+import cloneDeep from 'lodash/cloneDeep';
+import { translate } from '../../../i18n/translations';
+import { FormFields, FormFieldsArray } from '../../../types';
+
+/** Applies translations to label and placeholder */
+export const applyTranslation = (formFields: FormFields): FormFields => {
+  const newFormFields = { ...formFields };
+  for (const [name, options] of Object.entries(formFields)) {
+    const { label, placeholder } = options;
+
+    newFormFields[name] = {
+      ...options,
+      label: label ? translate<string>(label) : undefined,
+      placeholder: placeholder ? translate<string>(placeholder) : undefined,
+    };
+  }
+  return newFormFields;
+};
+
+/** Sorts formFields according to their `order`.  */
+export const sortFormFields = (formFields: FormFields): FormFieldsArray => {
+  return Object.entries(formFields)
+    .sort((a, b) => {
+      const orderA = a[1].order || Number.MAX_VALUE;
+      const orderB = b[1].order || Number.MAX_VALUE;
+      return orderA - orderB;
+    })
+    .filter((formFieldEntry) => formFieldEntry[1] !== undefined);
```

Comment:
I'm not sure if it's here or somewhere else, but the sorted fields are not showing correctly on the `sign-up-with-attributes` page. The `password` and `confirm_password` have been pushed to the bottom of the page. Usually the `username`, `password`, and `confirm_password` are always at the top.

<img width="498" alt="image" src="https://user-images.githubusercontent.com/65630/157536544-0039a768-b328-4825-b784-31c3f7ecc389.png">

Reply:
Oh man, good catch! The error stems from here: 

https://github.com/aws-amplify/amplify-ui/blob/d63f2b1086f5013cb57f405745bf9d91685825bc/packages/ui/src/helpers/authenticator/formFields/defaults.ts#L70-L77

The two password fields should be in the middle.

Reply:
Fixed here! f3024131440ffa1f8656932e53620d3b9ee0baaf

![image](https://user-images.githubusercontent.com/43682783/157542178-fe781278-9233-47d8-84e2-7e83e19792bc.png)